files/build_templates/docker_image_ctl.j2

```diff
@@ -169,18 +179,63 @@ start() {
 
     {%- if docker_container_name == "database" %}
     echo "Creating new {{docker_container_name}}$DEV container"
+    if [ -z "$DEV" ]; then
+        # if database_global exists in old_config, use it; otherwise use the default one in new image
+        if [ -f /etc/sonic/old_config/database_global.json ]; then
+            echo "Use database_global.json from old system..."
+            mv /etc/sonic/old_config/database_global.json /etc/sonic/
+        fi
+    fi
     # if database_config exists in old_config, use it; otherwise use the default one in new image
-    if [ -f /etc/sonic/old_config/database_config.json ]; then
+    if [ -f /etc/sonic/old_config/database_config$DEV.json ]; then
         echo "Use database_config.json from old system..."
-        mv /etc/sonic/old_config/database_config.json /etc/sonic/
+        mv /etc/sonic/old_config/database_config$DEV.json /etc/sonic/
     fi
     {%- else %}
     echo "Creating new {{docker_container_name}}$DEV container with HWSKU $HWSKU"
     {%- endif %}
 
+    # In Multi ASIC platforms the global database config file database_global.json will exist.
+    # Parse the file and get the include path for the database_config.json files used in
+    # various namesapces. The database_config paths are relative to the DIR of SONIC_DB_GLOBAL_JSON.
+    SONIC_DB_GLOBAL_JSON="/var/run/redis/sonic-db/database_global.json"
+    if [ -f "$SONIC_DB_GLOBAL_JSON" ]; then
+        db_configs=`/usr/bin/python -c "import sys; import os; import json; f=open(sys.argv[1]); \
+                    global_db_dir = os.path.dirname(sys.argv[1]); data=json.load(f); \
+                    print(\" \".join([os.path.normpath(global_db_dir+'/'+elem['include']).partition('sonic-db')[0]\
+                    for elem in data['INCLUDES'] if 'namespace' in elem])); f.close()" $SONIC_DB_GLOBAL_JSON`
+    fi
+
     if [ -z "$DEV" ]; then
         NET="host"
+
+        # For Multi-ASIC platform we have to mount the redis paths for database instances running in different
+        # namespaces, into the single instance dockers like snmp, pmon on linux host. These global dockers
+        # will need to get/set tables from databases in different namespaces.
+        # /var/run/redis0 ---> mounted as --> /var/run/redis0
+        # /var/run/redis1 ---> mounted as --> /var/run/redis1 .. etc
+        # The below logic extracts the base DIR's where database_config.json's for various namespaces exist.
+        # db_configs is a string of form "/var/run/redis0/ /var/run/redis1/ /var/run/redis2/"
+
+        {%- if docker_container_name != "database" %}
+        if [ -n "$db_configs" ]; then
+            for redis_dir in $db_configs
```

Comment:
can you change the name? like redis_dis_list?

Reply:
I have changed the name to redis_dir_list